horreum-web/src/domain/runs/DatasetComparison.tsx

```diff
@@ -213,28 +214,54 @@ type ViewComparisonProps = {
 
 function ViewComparison({headers, view, datasets, alerting}: ViewComparisonProps) {
     const [loading, setLoading] = useState(false)
-    const [rows, setRows] = useState<IRow[]>()
+    const [rows, setRows] = useState<IRow[]>([])
     const token = useSelector(tokenSelector)
+
     useEffect(() => {
         setLoading(true)
         Promise.all(datasets.map(ds => datasetApi.getSummary(ds.id, view.id)))
             .then(
                 summaries => {
-                    setRows(
-                        view.components.map(vc => ({
-                            cells: [
-                                vc.headerName,
-                                ...summaries.map(summary => {
-                                    const render = renderValue(
-                                        vc.render,
-                                        vc.labels.length == 1 ? vc.labels[0] : undefined,
-                                        token
-                                    )
-                                    return render(summary.view?.[vc.id], summary)
-                                }),
-                            ],
-                        }))
-                    )
+                    const rowsData: any[][] = view.components.map(vc => {
+                        const rowCells = [vc.headerName];
+                        summaries.forEach(summary => {
+                            const render = renderValue(vc.render, vc.labels.length === 1 ? vc.labels[0] : undefined, token);
+                            rowCells.push(render(summary.view?.[vc.id], summary));
+                        });
+                        return rowCells;
+                    });
```

Comment:
Is there a reason you removed the spread notation `[vc.headerName, ...summaries.map(` that was used before?
I think that would be easier to read but I'm curious if you found explicit iteration and mutation easier to read

Reply:
So while it might be pleasing to the eye I find it more straightforward to pass the function to render function. Also I need to capture this into a intermediate datastructure for further processing to detect numeric rows. Does that make sense why I didn't use spread notation ?

Reply:
```javascript
const rowsData: any[][] = view.components.map(vc => {
                        const rowCells = [vc.headerName];
                        summaries.forEach(summary => {
                            const render = renderValue(vc.render, vc.labels.length === 1 ? vc.labels[0] : undefined, token);
                            rowCells.push(render(summary.view?.[vc.id], summary));
                        });
                        return rowCells;
                    });
```
could be more succinctly expressed as 
```javascript
const rowsData: any[][] = view.components.map(vc => [
  vc.headerName,
  ...sumaries.map(summary => {
    const render = renderValue(vc.render, vc.labels.length === 1 ? vc.labels[0] : undefined, token);
    return render(summary.view?.[vc.id], summary)
  })
])  
```
They both produce the same `rowsData` content. I was curious if you had a preference for the explicit iteration with `forEach` and `push`. If you don't then would you please use the more succinct expression? I think it makes the code easier to read.